webapp/tests/forms/flows/test_unlock_code_revocation_flow.py

```diff
@@ -191,6 +191,61 @@ def test_if_user_exists_and_dob_correct_and_unlock_code_revocation_got_through_t
+            render_info, _ = self.flow._handle_specifics_for_step(
+                self.input_step, self.render_info_input_step, self.session_data)
+            self.assertEqual(self.failure_url, render_info.next_url)
+            
+    def test_if_user_exists_and_dob_incorrect_and_unlock_code_revocation_got_through_then_next_url_is_failure_step(self):
+        existing_idnr = '04452397687'
+        correct_dob = ['99', '99', '2021']
+        create_user(existing_idnr, '99.99.2021', '0000')
+        with self.app.test_request_context(method='POST',
+                                           data={'idnr': existing_idnr,
+                                                 'dob': correct_dob}):
+            render_info, _ = self.flow._handle_specifics_for_step(
+                self.input_step, self.render_info_input_step, self.session_data)
+            self.assertEqual(self.failure_url, render_info.next_url)
+
+    def test_if_user_exists_and_dob_in_the_future_and_unlock_code_revocation_got_through_then_next_url_is_failure_step(self):
+        existing_idnr = '04452397687'
+        correct_dob = ['1', '1', '2999']
+        create_user(existing_idnr, '01.01.2999', '0000')
+        with self.app.test_request_context(method='POST',
+                                           data={'idnr': existing_idnr,
+                                                 'dob': correct_dob}):
+            render_info, _ = self.flow._handle_specifics_for_step(
+                self.input_step, self.render_info_input_step, self.session_data)
+            self.assertEqual(self.failure_url, render_info.next_url)
+
+    def test_if_user_exists_and_dob_is_missing_and_unlock_code_revocation_got_through_then_next_url_is_failure_step(self):
+        existing_idnr = '04452397687'
+        correct_dob = ['', '', '']
+        create_user(existing_idnr, '01.01.2021', '0000')
+        with self.app.test_request_context(method='POST',
+                                           data={'idnr': existing_idnr,
+                                                 'dob': correct_dob}):
+            render_info, _ = self.flow._handle_specifics_for_step(
+                self.input_step, self.render_info_input_step, self.session_data)
+            self.assertEqual(self.failure_url, render_info.next_url)
+
+    def test_if_user_exists_and_dob_is_incomplete_and_unlock_code_revocation_got_through_then_next_url_is_failure_step(self):
+        existing_idnr = '04452397687'
+        correct_dob = ['01', '', '2021']
+        create_user(existing_idnr, '01.01.2021', '0000')
+        with self.app.test_request_context(method='POST',
+                                           data={'idnr': existing_idnr,
+                                                 'dob': correct_dob}):
+            render_info, _ = self.flow._handle_specifics_for_step(
+                self.input_step, self.render_info_input_step, self.session_data)
+            self.assertEqual(self.failure_url, render_info.next_url)
```

Comment:
Can you test this on the validator-level? so in webapp/tests/forms/validators.py?
I'm not sure if we also need the tests on a step-level? @JulianRoesner what do you think?

And I'd also suggest a test where the validator does not throw an error^^

Reply:
To explain this more: In this file there should only be tests concerning the logic of the unlock_code_revocation_flow - but we can test this way further down(on the validator) and thus do not have to mock the request etc.